Modules/_sha3/sha3module.c

```diff
@@ -262,7 +272,10 @@ SHA3_dealloc(SHA3object *self)
     if (self->lock) {
         PyThread_free_lock(self->lock);
     }
+
+    PyObject *tp = Py_TYPE(self);
```

Comment:
This line introduces a compiler warning, try:

```suggestion
    PyTypeObject *tp = Py_TYPE(self);
```